### ISSUE TITLE
How to properly use the scorer of the ACE2 task?

### ISSUE THREAD

**Issue by tedfeng424** [vishakhpk/iter-extrapolation#1]
I followed your script to train a scorer for the ACE2 dataset. How do I use it to measure the score of a sequence in the ACE2 dataset? I tried enabling `do_predict` on the evaluation dataset, but it returned the same score for all the sequences. Furthermore, how do you use the scorer to create the paired dataset?

**Comment by vishakhpk** [comment]
This code base is currently a work in progress.  The training script for the scorer is the [standard script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/text-classification) released by Huggingface for text classification, just adapted for this task. `do_predict` is from Huggingface and is used to make predictions just on the evaluation set. 

Once you train the scorer, you need to create mutations of the wildtype in order to create pairs using [this](https://github.com/vishakhpk/iter-extrapolation/blob/main/ace2/mlm_mutation.py) script and run inference on the newly created mutations using the trained scorer. I'm cleaning up the script to do this and will update it here soon. 

In the meantime, you can see the corresponding [script](https://github.com/vishakhpk/iter-extrapolation/blob/main/sentiment/denoise_and_score.py) in the sentiment task which runs inference using the scorer and then uses [this](https://github.com/vishakhpk/iter-extrapolation/blob/main/sentiment/create_json_file.py) to create the generator data. These can be minorly adapted to the ACE2 task as well.

**Comment by tedfeng424** [comment]
Thanks, I'll look into these

**Comment by tedfeng424** [comment]
I trained the ACE2 scoring model with the standard script, and try to run inferences following the codes in the sentiment task.  For example:

```
# sequences with ddg value ranging from -6 to 13
test_seq = ['S T I E E Q A K T F L D K F N H E K E D L F Y Q S S L A I W N Y N T N I T E E N V Q N M N F A Q D K W S A F L K E Q S T L A Q M Y P L Q E I Q N L T V K L F L Q A L M',
 'S T I E E Q A K T F L D K F N H E A E D L F Y Q S S L A S W F Y N T N I T E E N V Q N M M N A F D K W S A F L K E K S T L A Q M Y P L Q E I Y N L T V K L Q L Q A L Q',
 'S D I E E Q A K S F L D K F N H E A E D L F Y Q S S L A S W N Y N T N I T E E N V Q N M N N F G D K W S A F L K E L S T L A Q M Y P L Q E I Q N L T V K L Q L Q A L I',
 'S M I E E Q A K K F H P K F N H E A E D L F Y Q S S L A S M N Y N T N I T E E N V Q N M N N A G D K W S A F L K S Q S T L D Q M Y P L Q E I Q N L T V K L Q L Q A L Q',
 'S T I E E Q A K T F L D K F N H E A E D L F Y Q S S L A S W N Y N T N I T E E N P Q N M N N A G D K W S A F L K E Q Y T L A Q M Y P L Q E I Q N L T V K P Q L Q A L Q']

encodings = tokenizer(test_seq, padding=padding, max_length=max_seq_length, truncation=True, return_tensors='pt').to('cuda')
output = model(**encodings)

t = output['logits'].tolist()
flat_list = [item for sublist in t for item in sublist]
```
However, the scores/logits I got are extremely close to each other with some minor differences:
```
[0.9919768571853638,
 0.9919767379760742,
 0.9919768571853638,
 0.9919766783714294,
 0.9919769167900085]
```

Have you encounter similar problems, or is there any additional pre or post processings?

**Comment by vishakhpk** [comment]
598cb24
